### PR TITLE
Add a spellcheck into the CI

### PR DESCRIPTION
**Pull request by xgreenx**

### CODE REVIEW

File: .cargo/common.dic [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+TODO
+tokio
+transactional
+TTL
+tx
+TxPool/S
+txs
+uncompress
+uri
+UTXO/S
```
Comment (MitchTurner) [inline .cargo/common.dic:112]: I prefer `UTxO` tbo, so maybe add that too or instead?

Comment (MitchTurner) [reply .cargo/common.dic:112]: Hmm. Maybe that's not standard:
https://en.wikipedia.org/wiki/Unspent_transaction_output

Althought I prefer it, maybe just stick with the standard.

File: bin/fuel-core/src/cli/run/relayer.rs [hunk 2]
```diff
@@ -15,32 +15,32 @@ use std::str::FromStr;
 #[derive(Debug, Clone, Args)]
 pub struct RelayerArgs {
     /// Enable the Relayer. By default, the Relayer is disabled, even when the binary is compiled
-    /// with the "relayer" feature flag. Providing `--enable-relayer` will enable the relayer
+    /// with the "Relayer" feature flag. Providing `--enable-relayer` will enable the Relayer
```
Comment (MitchTurner) [inline bin/fuel-core/src/cli/run/relayer.rs:18]: I'm not sure that "relayer" should be capitalized. Maybe add it to the list in lowercase?

File: crates/fuel-core/src/executor.rs [hunk 3]
```diff
@@ -967,22 +967,22 @@ where
...
+                // ensure tx matches VM output exactly
                 if &tx != checked_tx.transaction() {
                     return Err(ExecutorError::InvalidTransactionOutcome {
                         transaction_id: tx_id,
                     })
                 }
             }
             ExecutionKind::DryRun | ExecutionKind::Production => {
-                // malleate the block with the resultant tx from the vm
+                // malleable the block with the resultant tx from the VM
```
Comment (MitchTurner) [inline crates/fuel-core/src/executor.rs:978]: "malleable" isn't a verb. I'm not sure what "malleate" means either. Maybe we choose a new word.

File: crates/fuel-core/src/executor.rs [hunk 4]
```diff
@@ -1485,7 +1485,7 @@ where
         }
     }
 
-    /// Log a VM backtrace if configured to do so
+    /// Log a VM backtraces if configured to do so
```
Comment (MitchTurner) [inline crates/fuel-core/src/executor.rs:1482]: I think "backtrace" was more correct.

File: crates/fuel-core/src/executor/ports.rs [hunk 5]
```diff
@@ -51,8 +51,8 @@ pub trait RelayerPort {
 
 #[cfg(test)]
 /// For some tests we don't care about the actual
-/// implementation of the RelayerPort
-/// and using a passthrough is fine.
+/// implementation of the `RelayerPort`
+/// and using a pass through is fine.
```
Comment (MitchTurner) [inline crates/fuel-core/src/executor/ports.rs:55]: "pass-through"

File: crates/fuel-core/src/schema/block.rs [hunk 6]
```diff
@@ -69,7 +69,7 @@ type CoreConsensus = fuel_core_types::blockchain::consensus::Consensus;
 
 #[derive(SimpleObject)]
 pub struct Genesis {
-    /// The chain configs define what consensus type to use, what settlement layer to use,
+    /// The chain config define what consensus type to use, what settlement layer to use,
```
Comment (MitchTurner) [inline crates/fuel-core/src/schema/block.rs:72]: config "defines"

File: crates/services/producer/src/block_producer.rs [hunk 7]
```diff
@@ -88,14 +88,14 @@ where
...
         //  - get previous block info (hash, root, etc)
-        //  - select best da_height from relayer
-        //  - get available txs from txpool
+        //  - select best `da_height` from Relayer
+        //  - get available txs from TxPool
         //  - select best txs based on factors like:
         //      1. fees
         //      2. parallel throughput
-        //  - Execute block with production mode to correctly malleate txs outputs and block headers
+        //  - Execute block with production mode to correctly malleable txs outputs and block headers
```
Comment (MitchTurner) [inline crates/services/producer/src/block_producer.rs:96]: Here's another "malleable" being used as a verb.

File: crates/services/producer/src/block_producer.rs [hunk 8]
```diff
@@ -144,8 +144,8 @@ where
         // The dry run execution should use the state of the blockchain based on the
         // last available block, not on the upcoming one. It means that we need to
         // use the same configuration as the last block -> the same DA height.
-        // It is deterministic from the result perspective, plus it is more performant
-        // because we don't need to wait for the relayer to sync.
+        // It is deterministic from the result perspective, plus it is more performer
```
Comment (MitchTurner) [inline crates/services/producer/src/block_producer.rs:147]: What was wrong with "performant"? "performer" is wrong.

File: crates/services/producer/src/block_producer.rs [hunk 9]
```diff
@@ -201,7 +201,7 @@ where
         let best_height = self.relayer.wait_for_at_least(&previous_da_height).await?;
         if best_height < previous_da_height {
             // If this happens, it could mean a block was erroneously imported
-            // without waiting for our relayer's da_height to catch up to imported da_height.
+            // without waiting for our `Relayer`'s `da_height` to catch up to imported `da_height`.
```
Comment (MitchTurner) [inline crates/services/producer/src/block_producer.rs:204]: We use backticks here (`) but not in other places.

File: crates/client/assets/schema.sdl [hunk 10]
```diff
@@ -258,7 +258,7 @@ type DependentCost {
 
 input ExcludeInput {
 	"""
-	Utxos to exclude from the selection.
+	UTXO to exclude from the selection.
```
Comment (MitchTurner) [inline crates/client/assets/schema.sdl:261]: "UTXOs"

File: crates/services/relayer/src/service.rs [hunk 11]
```diff
@@ -252,13 +252,13 @@ impl<D> SharedState<D> {
     /// Wait for the [`Task`] to be in sync with
     /// the data availability layer.
     ///
-    /// Yields until the relayer reaches a point where it
+    /// Yields until the Relayer reaches a point where it
```
Comment (bvrooman) [inline crates/services/relayer/src/service.rs:255]:
```suggestion
    /// Yields until the Relayer reaches a point where it is
```

File: crates/services/relayer/src/test_helpers/middleware.rs [hunk 12]
```diff
@@ -165,7 +165,7 @@ impl Default for MockMiddleware {
 #[derive(Error, Debug)]
 /// Thrown when an error happens at the Nonce Manager
 pub enum MockMiddlewareError {
-    /// Thrown when the internal middleware errors
+    /// Thrown when the internal middle ware errors
```
Comment (MitchTurner) [inline crates/services/relayer/src/test_helpers/middleware.rs:168]: I think "middleware" is correct.

File: crates/services/src/service.rs [hunk 13]
```diff
@@ -66,13 +66,13 @@ pub trait Service {
 /// Trait used by `ServiceRunner` to encapsulate the business logic tasks for a service.
 #[async_trait::async_trait]
 pub trait RunnableService: Send {
-    /// The name of the runnable service, used for namespacing error messages.
+    /// The name of the runnable service, used for name spacing error messages.
```
Comment (MitchTurner) [inline crates/services/src/service.rs:69]: "namespacing" is correct

File: crates/services/relayer/src/test_helpers/middleware.rs (hunk 12 above)
Comment (bvrooman) [inline crates/services/relayer/src/test_helpers/middleware.rs:168]: This should still be `"middleware"`

File: crates/services/src/service.rs (hunk 13 above)
Comment (bvrooman) [inline crates/services/src/service.rs:69]: This should still be `"namespacing"`

File: crates/services/sync/src/ports.rs [hunk 14]
```diff
@@ -64,7 +64,7 @@ pub trait PeerToPeerPort {
 pub trait ConsensusPort {
     /// Check if the given sealed block header is valid.
     fn check_sealed_header(&self, header: &SealedBlockHeader) -> anyhow::Result<bool>;
-    /// await for this DA height to be sync'd.
+    /// await for this DA height to be synced.
```
Comment (bvrooman) [inline crates/services/sync/src/ports.rs:67]:
```suggestion
    /// await for this DA height to be synchronized.
```

File: crates/services/txpool/src/txpool.rs [hunk 15]
```diff
@@ -227,8 +227,8 @@ where
         tx_status_sender: &TxStatusChange,
         txs: Vec<Checked<Transaction>>,
     ) -> Vec<anyhow::Result<InsertionResult>> {
-        // Check if that data is okay (witness match input/output, and if recovered signatures ara valid).
-        // should be done before transaction comes to txpool, or before it enters RwLocked region.
+        // Check if that data is okay (witness match input/output, and if recovered signatures  valid).
```
Comment (bvrooman) [inline crates/services/txpool/src/txpool.rs:230]:
```suggestion
        // Check if that data is okay (witness matches input/output, and recovered signatures are valid).
```

File: crates/types/src/blockchain/block.rs [hunk 16]
```diff
@@ -99,7 +99,7 @@ impl Block<Transaction> {
 }
 
 impl<T> Block<T> {
-    /// Destructure into the inner types.
+    /// Destructs into the inner types.
```
Comment (bvrooman) [inline crates/types/src/blockchain/block.rs:102]: This should still be "Destructure"

File: crates/types/src/blockchain/block.rs [hunk 17]
```diff
@@ -41,7 +41,7 @@ pub type CompressedBlock = Block<TxId>;
 
 /// Fuel block with all transaction data included
 /// but without any data generated.
-/// This type can be created with unexecuted
+/// This type can be created with non-executed
```
Comment (bvrooman) [inline crates/types/src/blockchain/block.rs:44]: This should still be "unexecuted"

File: .cargo/common.dic [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+(
+)
+*
+/
+@dev
+^
++
+<
+=
+>
```
Comment (bvrooman) [inline .cargo/common.dic:13]: Why are these included in the dictionary?

File: .cargo/common.dic [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+Jemalloc
+JSON
+Kademlia
+keygen
+keypair
+libp2p
+lifecycle
+MDNS
+malleate
+merkle
```
Comment (bvrooman) [inline .cargo/common.dic:66]:
```suggestion
Merkle
```

**Review by bvrooman — COMMENTED** [review]
A couple of concerns:
- There are currently a number of things added to the dictionary that are either real words that probably don't need to be there, or fake words/shorthands that probably shouldn't be there
- This check can only do so much, and we should be more attentive to how we write our comments in general. And as part of the PR review process, we should be catching spelling and grammar mistakes at that time
- Some other acronyms like "db" should arguably be flagged, and we should instead use something like "DB". It does not seem like this check is catching that.

File: .cargo/common.dic (hunk 18 above)
Comment (xgreenx) [reply .cargo/common.dic:13]: Otherwise, we can't use formulas in the comments=)

File: crates/fuel-core/src/executor.rs (hunk 3 above)
Comment (Dentosal) [reply crates/fuel-core/src/executor.rs]: I would suggest "mutate" here

File: crates/services/producer/src/block_producer.rs (hunk 7 above)
Comment (Dentosal) [reply crates/services/producer/src/block_producer.rs]: Maybe "modify" instead?

**Comment by Dentosal** [comment]
I don't think this is worth doing. Too many false positives, and the impact of typos is really low.

File: crates/services/producer/src/block_producer.rs (hunk 7 above)
Comment (Voxelot) [reply crates/services/producer/src/block_producer.rs]: We could go with modify, but malleate carries some context from the bitcoin world which is useful conceptually.

**Comment by xgreenx** [comment]
> I don't think this is worth doing. Too many false positives, and the impact of typos is really low.

There are indeed a lot of false positives because we need to set the dictionary that works with our style of comments=) And it is almost done because we have already received reviews from everyone.

It is nice to have a spellchecker for the open-source project. Because it is already done(we spent some time adding it and reviewing it), we can integrate it(plus it is an optional part of the CI). But I'm also okay to close this pull request if no one approves it by the end of this week=)

**Comment by xgreenx** [comment]
No one approved, so I'm closing this PR=)
